### ISSUE TITLE
Additional Callbacks

### ISSUE THREAD

**Issue by tomberek** [notedit/gst#5]
I was going to give this library a try and implement this in Go:
http://blog.nirbheek.in/2018/02/gstreamer-webrtc.html

But quickly ran into the limited callback functions and private callbackID on Element. I'm experimenting with adding things like SetOnNegotiationNeededCallback and SetOnICECandidateCallback, or something a little more general allowing arbitrary callbacks, but I thought I'd register this issue to see if adding something like this upstream would be possible.

**Comment by notedit** [comment]
there is a on-padadded callback.

https://github.com/notedit/gst/blob/e023dcb2fc1b945f3ad3e6727ed68956d00490c3/element.go#L270

you can try something similar.

**Comment by notedit** [comment]
`something a little more general allowing arbitrary callbacks`  is a better choice i think.

**Comment by tomberek** [comment]
Yes, I was going to implement the same thing just with a different string, but I cannot access the callbackStore or private fields like Element.callbackID.

btw: i'm also taking a look at media-server-go, trying to see where i can integrate easier.

**Comment by notedit** [comment]
Thanks,  if you can make a pr.

**Comment by tomberek** [comment]
Arbitrary callbacks signals and sending go functions through for gstreamer to call is proving challenging. Was trying to use this as a reference: https://eli.thegreenplace.net/2019/passing-callbacks-and-pointers-to-cgo/
